server/activity/activity.controller.js

```diff
@@ -59,6 +62,16 @@ function clone({ activity, body }, res) {
   });
 }
 
+function getPreview({ activity }, res) {
+  return fetchContainer(activity)
+    .then(data => request.post(previewUrl, { json: true, body: data }))
+    .then(data => res.json(resolve(previewUrl, data.body.data)));
```

Comment:
```suggestion
    .then(data => res.json(`${new URL(data.body.data, previewUrl)}`);
```
However don't return plain string, make it part of an object:
```js
{ location: ... }
```